src/components/Truncate/Truncate.tsx

```diff
@@ -70,24 +75,25 @@ export class Truncate extends React.PureComponent<
       /* istanbul ignore next */
       if (!this.node || !this.contentNode) return false
 
-      // 1. Normalizes the display to allow for calculation
-      // TODO: fix typescript complains
-      // @ts-ignore
-      this.contentNode.style.display = 'initial'
-      // 2. Calculate the differences
       const isContentTruncated =
-        // TODO: fix typescript complains
-        // @ts-ignore
-        this.contentNode.offsetWidth > this.node.offsetWidth
-      // 3. Resets the display
-      // TODO: fix typescript complains
-      // @ts-ignore
-      this.contentNode.style.display = null
+        props.splitter && props.showTooltipOnTruncate
+          ? this.isSplitContentTruncated(this.contentNode, this.node)
+          : // TODO: fix typescript complains
+            // @ts-ignore
+            this.contentNode.scrollWidth > this.node.offsetWidth
 
       return isContentTruncated
     }
   }
 
+  isSplitContentTruncated = (contentNode: any, node: any): boolean => {
```

Comment:
Tried wrapping this in a guard as well so I could use `HTMLElement` as a type but it still complained. No idea what's going on.